### PR TITLE
Add RBAC support

### PR DESCRIPTION
**Pull request by jgroh9**
# What was done

- Add new script to create RBAC users
- Add sample configuration file for RBAC user creation
- Add documentation

### CODE REVIEW

**Review by brantburnett — CHANGES_REQUESTED** [review]
Since RBAC users are at the cluster level, and the role definition in the JSON file must include the bucket name, I'm wondering if we should make just a single file with all RBAC users at the root of startup, rather than a separate file per bucket. This would support one user with access to more than one bucket and be a bit clearer.

**Comment by jgroh9** [comment]
Requested changes have been implemented and I've merged main to resolve all conflicts

**Comment by jgroh9** [comment]
@brantburnett Any chance you'll be able to review this PR this week?

File: scripts/create-rbac-users.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  do
+    rbacBucketName=$(echo $rbacSettings | jq -r '.["bucket"]')
+    rbacName=$(echo $rbacSettings | jq -r '.["rbacName"]')
+    rbacUsername=$(echo $rbacSettings | jq -r '.["rbacUsername"]')
+    rbacPassword=$(echo $rbacSettings | jq -r '.["rbacPassword"]')
+    rbacRoles=$(echo $rbacSettings | jq -r '.roles | join(",")')
+
+    echo "Bucket Name = $rbacBucketName"
+
+    if [[ ! -z $rbacBucketName ]]; then
```
Comment (brantburnett) [inline scripts/create-rbac-users.sh:16]: My only remaining concern is the presence of bucketName on the JSON doc and validating that the folder exists. I see two concerns. First, you can make a bucket in buckets.json without having a bucket folder that defines indexes/models. Second, it's theoretically possible for a user to not have any specific bucket association and be purely admin in nature.

Do you have a specific use case in mind for this property?

**Review by brantburnett — CHANGES_REQUESTED** [review]

File: scripts/create-rbac-users.sh (hunk 1 above)
Comment (jgroh9) [reply scripts/create-rbac-users.sh:16]: I see your point. I'll remove the bucket property from the JSON and the rbacBucketName variable

Comment (jgroh9) [reply scripts/create-rbac-users.sh]: I've made the requested changes in this [commit](https://github.com/brantburnett/couchbasefakeit/pull/25/commits/b5f67259c004ee7f9db64316639ef0b4c35bb9e3)

**Review by brantburnett — APPROVED** [review]

**Comment by brantburnett** [comment]
Thanks so much, looks like it works great.

**Pull request merged**